### PR TITLE
feat(mc): #2017 Add blocking to snippets.js

### PR DESCRIPTION
**Pull request by k88hudson**
This adds blocking functionality for snippets that can be used by the payload JS

### CODE REVIEW

File: system-addon/content-src/lib/snippets.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ class SnippetsMap extends Map {
     return this._dbTransaction(db => db.put(value, key));
   }
 
-  delete(key, value) {
+  delete(key, valuatoe) {
```
Comment (sarracini) [inline system-addon/content-src/lib/snippets.js]: i think this is a typo?

File: system-addon/content-src/lib/snippets.js [hunk 2]
```diff
@@ -32,6 +32,28 @@ class SnippetsMap extends Map {
...
+   * @return {Promise}      Resolves when the id has been written to indexedDB,
+   *                        or immediately if the snippetMap is not connected
+   */
+  async blockSnippetById(id) {
+    if (!id) {
+      return;
+    }
+    let blockList = this.blockList;
+    if (!blockList.includes(id)) {
+      blockList = [id].concat(blockList);
```
Comment (sarracini) [inline system-addon/content-src/lib/snippets.js]: are you not able to just push id onto blockList?

File: system-addon/content-src/lib/snippets.js (hunk 1 above)
Comment (k88hudson) [inline system-addon/content-src/lib/snippets.js]: looool

File: system-addon/content-src/lib/snippets.js (hunk 2 above)
Comment (k88hudson) [inline system-addon/content-src/lib/snippets.js]: Yeah I could, I was in no-mutate react mode I guess

File: system-addon/content-src/lib/snippets.js (hunk 1 above)
Comment (k88hudson) [inline system-addon/content-src/lib/snippets.js]: Actually, there isn't a value for delete anyway...

**Comment by as-pine-proxy** [comment]
[Treeherder: b4936ab8d8039e3172a48dc7d879f6eca4ddf21e](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=b4936ab8d8039e3172a48dc7d879f6eca4ddf21e)

**Pull request merged**